src/bench/connectblock.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2025 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <addresstype.h>
+#include <bench/bench.h>
+#include <interfaces/chain.h>
+#include <kernel/cs_main.h>
+#include <script/interpreter.h>
+#include <sync.h>
+#include <test/util/setup_common.h>
+#include <validation.h>
+
+#include <cassert>
+#include <vector>
+
+/*
+ * Creates a test block containing transactions with the following properties:
+ * - Each transaction has the same number of inputs and outputs
+ * - All Taproot inputs use simple key path spends (no script path spends)
+ * - All signatures use SIGHASH_ALL (default sighash)
+ * - Each transaction spends all outputs from the previous transaction
+ */
+CBlock CreateTestBlock(
+    TestChain100Setup& test_setup,
+    const std::vector<CKey>& keys,
+    const std::vector<CTxOut>& outputs,
+    int num_txs = 1000)
+{
+    Chainstate& chainstate{test_setup.m_node.chainman->ActiveChainstate()};
+
+    const WitnessV1Taproot coinbase_taproot{XOnlyPubKey(test_setup.coinbaseKey.GetPubKey())};
+
+    // Create the outputs that will be spent in the first transaction of the test block
+    // Doing this in a separate blocks excludes the validation of it's inputs from the benchmark
+    auto coinbase_to_spend{test_setup.m_coinbase_txns[0]};
+    const auto [first_tx, _]{test_setup.CreateValidTransaction(
+        std::vector{coinbase_to_spend},
+        std::vector{COutPoint(coinbase_to_spend->GetHash(), 0)},
+        chainstate.m_chain.Height() + 1, keys, outputs, std::nullopt, std::nullopt)};
+    const auto test_block_parent_coinbase{GetScriptForDestination(coinbase_taproot)};
+    test_setup.CreateAndProcessBlock(std::vector{first_tx}, test_block_parent_coinbase, &chainstate);
+
+    std::vector<CMutableTransaction> txs;
+    txs.reserve(num_txs);
```

Comment:
not blocking, just a question: here and below, why `reserve()` for vectors that are outside of the benchmark's path?

Reply:
I understand that the use of `reserve` here is not critical, but, I generally use `reserve()` whenever I know the size of the vector beforehand in line with best practices

Reply:
It's good general practice and it documents the code as well, e.g:
```
    std::vector<CKey> keys{coinbaseKey};
    keys.reserve(num_taproot + num_nontaproot + 1);
````